tests/test_hf_api.py

```diff
@@ -2427,6 +2427,52 @@ def test_get_repo_discussion(self):
             list([d.num for d in discussions_generator]), [self.discussion.num, self.pull_request.num]
         )
 
+    def test_get_repo_discussion_by_type(self):
+        discussions_generator = self._api.get_repo_discussions(repo_id=self.repo_id, discussion_type="pull_request")
+        self.assertIsInstance(discussions_generator, types.GeneratorType)
+        self.assertListEqual(list([d.num for d in discussions_generator]), [self.pull_request.num])
+
+        discussions_generator = self._api.get_repo_discussions(repo_id=self.repo_id, discussion_type="discussion")
+        self.assertIsInstance(discussions_generator, types.GeneratorType)
+        self.assertListEqual(list([d.num for d in discussions_generator]), [self.discussion.num])
+
+        discussions_generator = self._api.get_repo_discussions(repo_id=self.repo_id, discussion_type="all")
+        self.assertIsInstance(discussions_generator, types.GeneratorType)
+        self.assertListEqual(
+            list([d.num for d in discussions_generator]), [self.discussion.num, self.pull_request.num]
+        )
+
+    @unittest.skip(
+        "__DUMMY_TRANSFORMERS_USER__ is not a valid username anymore. This test fails the Hub input validation in consequence."
+    )
+    def test_get_repo_discussion_by_author(self):
+        discussions_generator = self._api.get_repo_discussions(repo_id=self.repo_id, author="no-discussion")
+        self.assertIsInstance(discussions_generator, types.GeneratorType)
+        self.assertListEqual(list([d.num for d in discussions_generator]), [])
+
+        discussions_generator = self._api.get_repo_discussions(repo_id=self.repo_id, author=USER)
+        self.assertIsInstance(discussions_generator, types.GeneratorType)
+        self.assertListEqual(
+            list([d.num for d in discussions_generator]), [self.discussion.num, self.pull_request.num]
+        )
```

Comment:
```suggestion
    def test_get_repo_discussion_by_author(self):
        discussions_generator = self._api.get_repo_discussions(repo_id=self.repo_id, author="unknown")
        self.assertIsInstance(discussions_generator, types.GeneratorType)
        self.assertListEqual(list([d.num for d in discussions_generator]), [])
```

Let's not add a failing test. This should be fine